File: C++/product.cpp
#include "header.hh"

/* Constructor. */

// Empty constructor.
Product::Product()
{

}

// Constructor with product's data (will it be used?).
Product::Product(int price, string idProduct)
{
    this->price = price;
    this->idProduct = idProduct;
}

/* Setter and Getter. */

// Set price attribute.
void Product::setPrice(int price)
{
    this->price = price;
}

// Set product's ID attribute.
void Product::setIDProduct(string idProduct)
{
    this->idProduct = idProduct;
}

// Get price attribute.
int Product::getPrice()
{
    return price;
}

// Get product's ID attribute.
string Product::getIDProduct()
{
    return idProduct;
}

/* Destructor. */

Product::~Product()
{
    
}

File: Java/Product.java
// Class Product, treated as superclass.
public class Product
{
    // Product attributes.
    private int price;
    private String idProduct;
    
    /* Constructor. */

    // Empty constructor.
    public Product()
    {

    }

    // Constructor with product's data (will it be used?).
    public Product(int price, String idProduct)
    {
        this.price = price;
        this.idProduct = idProduct;
    }

    /* Setter and Getter. */

    // Set price attribute.
    public void setPrice(int price)
    {
        this.price = price;
    }

    // Set product's ID attribute.
    public void setIDProduct(String idProduct)
    {
        this.idProduct = idProduct;
    }

    // Get price attribute.
    public int getPrice()
    {
        return this.price;
    }

    // Get product's ID attribute.
    public String getIDProduct()
    {
        return this.idProduct;
    }
}
